[INKLB-006] network: client session: add manual websocket client session (from specific port)

sonofR9 opened the pull request:
https://trello.com/c/mLaxRSWf/40-client-back-session-с-определённого-порта

Зависит от INKLB-004 https://github.com/cpp-park-vk-education/2023_1_The_Loopers/pull/14 (там общие концепты определены)

sonofR9 committed:
INKLB-006 add manual websocket client

sonofR9 committed:
Merge branch 'dev/a.novak/network' of github.com:cpp-park-vk-education/2023_1_The_Loopers into dev/a.novak/manual_client_session

sonofR9 committed:
INKLB-007 after merge

sonofR9 committed:
Merge branch 'dev/a.novak/network' of github.com:cpp-park-vk-education/2023_1_The_Loopers into dev/a.novak/manual_client_session

sonofR9 committed:
Merge branch 'dev/a.novak/network' of github.com:cpp-park-vk-education/2023_1_The_Loopers into dev/a.novak/manual_client_session

sonofR9 committed:
Merge branch 'dev/a.novak/network' of github.com:cpp-park-vk-education/2023_1_The_Loopers into dev/a.novak/manual_client_session

sonofR9 committed:
INKLB-005 FIX probably right way to close

Arugaf commented on utils/client_connector/src/manual_websocket_client_session.cpp:
```diff
@@ -0,0 +1,212 @@
+void WebsocketClientSession<DoOnConnectType, DoOnRead, DoOnWrite, DoOnClose>::RunAsync(const std::string& host,
+                                                                                       unsigned short port)
+{
+    m_host = host;
+    m_host += ':' + std::to_string(port);
```
Почему не в одну строчку?

Вообще это бы вынес в отдельную функцию в анонимном неймспейсе.

Arugaf reviewed (changes requested):
И тут ci + обычные замечания (особенно про доку :rofl:)

Ссылку на таску в человекочитаемый формат.

sonofR9 committed:
Merge branch 'dev/a.novak/network' of github.com:cpp-park-vk-education/2023_1_The_Loopers into dev/a.novak/manual_client_session

sonofR9 committed:
INKLB-006 REF after review

sonofR9 committed:
Merge branch 'dev/a.novak/network' of github.com:cpp-park-vk-education/2023_1_The_Loopers into dev/a.novak/manual_client_session

kingjoram commented on utils/client_connector/CMakeLists.txt:
```diff
@@ -2,6 +2,7 @@ add_library(${CLIENT_NETWORK} STATIC "")
 
 target_sources(${CLIENT_NETWORK} PRIVATE
     "src/trash.cpp"
+    "src/manual_websocket_client_session.cpp"
```
не надо ли уже удалить мусорку?

sonofR9 replied:
Надо, удалю)

kingjoram commented on utils/client_connector/src/manual_websocket_client_session.cpp:
```diff
@@ -0,0 +1,240 @@
+        return;
+    }
+}
+} // namespace inklink::client_connector
```
в конце пустую строку потерял

kingjoram commented on utils/client_connector/include/manual_websocket_client_session.h:
```diff
@@ -0,0 +1,84 @@
+    ConnectCallback m_connectCallback;
+    ReadCallback m_readCallback;
+    WriteCallback m_writeCallback;
+    CloseCallback m_closeCallback;
+};
+} // namespace inklink::client_connector
+
+#endif // _MANUAL_WEBSOCKET_CLIENT_SESSION_H_
```
пустая строка в конце

kingjoram replied:
а еще не знаю, нужен ли отступ перед endif

sonofR9 replied:
Не уверен, но такого инклуда быть не должно

kingjoram reviewed (commented):
ну, от меня как обычно мелкие замечания

sonofR9 committed:
INKLB-005 FIX constructor default params

sonofR9 committed:
Merge branch 'dev/a.novak/network' of github.com:cpp-park-vk-education/2023_1_The_Loopers into dev/a.novak/manual_client_session

sonofR9 committed:
INKLB-006 atomic, websocket stream is pointer

It has deleted operator=
But I need to assign it only after socket has established connection

sonofR9 committed:
INKLB-006 FIX kClientName

For some reason does not want to concatenate with boost macros

Arugaf commented on utils/client_connector/include/manual_websocket_client_session.h:
```diff
@@ -0,0 +1,82 @@
+    void OnWrite(error_code ec, std::size_t bytes_transferred);
+    void OnRead(error_code ec, std::size_t bytes_transferred);
+    void OnClose(error_code ec);
+
+    boost::asio::ip::tcp::socket m_socket;
```
Модификатор доступа на поля.

sonofR9 replied:
Сокет здесь я вручную создаю, а потом std::move вызываю, так что const не получится.
С connectCallback и т.д. не уверен, честно говоря, будет ли оно норма работать с const
 А на остальное, вроде, тоже не выходит.

Arugaf replied:
Модификатор доступа - это `private`/`protected`/`public`.

sonofR9 replied:
А, туплю
Но там, вроде, всё private. Либо я не понял что-то((

Arugaf commented on utils/client_connector/src/manual_websocket_client_session.cpp:
```diff
@@ -0,0 +1,241 @@
+namespace websocket = beast::websocket;
+using error_code = boost::system::error_code;
+
+constexpr const char* kClientName = "inklink-client";
```
Excess const.

Arugaf replied:
`constexpr std::string_view ...`

sonofR9 committed:
Merge branch 'dev/a.novak/network' of github.com:cpp-park-vk-education/2023_1_The_Loopers into dev/a.novak/manual_client_session

sonofR9 committed:
template class to header

sonofR9 committed:
Template => in header

sonofR9 committed:
INKLB-006 remove cpp

sonofR9 committed:
INKLB-006 Add access modifiers

Arugaf reviewed (approved):
LGTM
